cranelift/codegen/src/timing.rs

```diff
@@ -273,14 +278,18 @@ mod enabled {
 #[cfg(not(feature = "timing"))]
 mod disabled {
     use super::{DefaultProfiler, Pass, Profiler};
-    use std::any::Any;
-    use std::boxed::Box;
+    use alloc::boxed::Box;
+    use core::any::Any;
 
     impl Profiler for DefaultProfiler {
         fn start_pass(&self, _pass: Pass) -> Box<dyn Any> {
             Box::new(())
```

Comment:
This should probably return a private type to ensure it can't be downcasted.

Reply:
okay maybe this is stupid of me, but why would that be an issue?

Reply:
It is not that much of an issue, but if in the future we were to add an actually working profiler (eg by using atomics and aggregating statistics for all threads), it would technically be a breaking change to change the type stored in the `Box<dyn Any>` as now `.downcast::<()>()` for consumers would fail. I don't think anyone will write code like that though.